### PR TITLE
feat: resolver concurrency

### PR DESCRIPTION
**Pull request by schwma**
According to the [GraphQL spec](https://spec.graphql.org/October2021/#sec-Normal-and-Serial-Execution) queries should be executed in parallel and mutations should be executed serially. This PR addresses that.

Note: the spec refers to top-level fields, which would be services in our case. Instead we apply the parallel or serial execution to the resolvers of the entities that are exposed within each service instead.

### CODE REVIEW

File: lib/resolvers/root.js [hunk 1]
```diff
@@ -16,7 +16,7 @@ const _wrapResolver = (service, resolver) =>
         const entity = service.entities[fieldName]
         const responseKey = field.alias?.value || fieldName
 
-        response[responseKey] = resolver(service, entity, field)
+        response[responseKey] = parallel ? resolver(service, entity, field) : await resolver(service, entity, field)
```
Comment (David-Kunz) [inline lib/resolvers/root.js:19]: The promises are awaited via Promise.all by the GraphQL framework?

Comment (David-Kunz) [reply lib/resolvers/root.js:19]: Or is that done by you?

Comment (David-Kunz) [reply lib/resolvers/root.js:19]: I don't think that works

Comment (David-Kunz) [reply lib/resolvers/root.js:19]:
```js
await Promise.all(enrichedFieldNodes.map(fieldNode => Promise.all(fieldNode.selectionSet.selections.map(async field => {
  const fieldName = field.name.value
  const entity = service.entities[fieldName]
  const responseKey = field.alias?.value || fieldName
  response[responseKey] = await resolver(service, entity, field)
}))))
```

**Comment by schwma** [comment]
I think this PR still needs a bit of work. For example, it could benefit from some tests that contain responses consisting of both resolved and rejected promises for queries and mutations. Therefore changed PR to draft for now.

File: lib/resolvers/root.js (hunk 1 above)
Comment (schwma) [reply lib/resolvers/root.js]: Adapted solution as discussed in our call

**Review by David-Kunz — APPROVED** [review]

File: test/resources/concurrency/srv/concurrency.js [hunk 2]
```diff
@@ -0,0 +1,29 @@
+
+const cds = require('@sap/cds')
+const util = require('util')
+const sleep = util.promisify(setTimeout)
+
+class ConcurrencyService extends cds.ApplicationService { init(){
+
+  const logSleep = async (duration, message) => {
+    console.log('BEGIN ' + message)
```
Comment (David-Kunz) [inline test/resources/concurrency/srv/concurrency.js:9]: needed?

File: test/tests/concurrency.test.js [hunk 3]
```diff
@@ -0,0 +1,211 @@
...
+
+  const { axios, POST } = cds.test(path.join(__dirname, '../resources/concurrency'))
+  // Prevent axios from throwing errors for non 2xx status codes
+  axios.defaults.validateStatus = false
+
+  describe('execution order of query and mutation resolvers', () => {
+    let _log
+
+    beforeEach(() => {
+      _log = jest.spyOn(console, 'log')
```
Comment (David-Kunz) [inline test/tests/concurrency.test.js:14]: Ah okay. Ignore my other comment.

**Pull request merged**